lineage.go

```diff
@@ -268,3 +279,23 @@ type unaryConvLineage[T Assignee] struct {
 func (lin *unaryConvLineage[T]) TypedSchema() TypedSchema[T] {
 	return lin.tsch
 }
+
+func IsAppendOnly(oldLineage Lineage, newLineage Lineage) bool {
```

Comment:
i think this is the signature we want. Even if we can't convey much in the error message right now, it'll be crucial that a failure here is able to provide line-level feedback on where the problem was.

```suggestion
func IsAppendOnly(old, new Lineage) error {
```